src/frontend/src/lib/utils/accessMethods.ts

```diff
@@ -53,14 +53,40 @@ export const getLastUsedAccessMethod = (
   })[0];
 };
 
-const hasOrigin = (
+/**
+ * Extract the origin from an AuthnMethodData's metadata
+ */
+export const getOrigin = (
   accessMethod: AuthnMethodData,
-  origin: string[],
-): boolean => {
+): string | undefined => {
   const metadataEntry = accessMethod.metadata.find(([key]) => key === "origin");
   const metadataValue = metadataEntry?.[1];
   if (nonNullish(metadataValue) && "String" in metadataValue) {
-    return origin.some((o) => isSameOrigin(o, metadataValue.String));
+    return metadataValue.String;
+  }
+  return undefined;
+};
+
+/**
+ * Check if there are multiple unique origins across authentication methods
+ */
+export const haveMultipleOrigins = (
+  authnMethods: AuthnMethodData[],
+): boolean => {
+  const origins = new Set<string | undefined>();
+  for (const method of authnMethods) {
+    origins.add(getOrigin(method));
+  }
+  return origins.size > 1;
```

Comment:
Nit, this could be simplified:

```suggestion
  return new Set(authnMethods.map(getOrigin)).size > 1;
```